syntax = "proto3";

package jogasaki.proto.sql.status;

option java_package = "com.tsurugidb.sql.proto";
option java_outer_classname = "SqlStatus";

enum Status {
  OK = 0;

  NOT_FOUND = 1;
  ALREADY_EXISTS = 2;
  USER_ROLLBACK = 3;

  ERR_UNKNOWN = -1;
  ERR_IO_ERROR = -2;
  ERR_PARSE_ERROR = -3;
  ERR_TRANSLATOR_ERROR = -4;
  ERR_COMPILER_ERROR = -5;
  ERR_INVALID_ARGUMENT = -6;
  ERR_INVALID_STATE = -7;
  ERR_UNSUPPORTED = -8;
  ERR_USER_ERROR = -9;
  ERR_ABORTED = -10;
  ERR_SERIALIZATION_FAILURE = -11;
  ERR_NOT_FOUND = -12;
  ERR_ALREADY_EXISTS = -13;
  ERR_INCONSISTENT_INDEX = -14;
  ERR_TIME_OUT = -15;
  ERR_INTEGRITY_CONSTRAINT_VIOLATION = -16;
  ERR_EXPRESSION_EVALUATION_FAILURE = -17;
  ERR_UNRESOLVED_HOST_VARIABLE = -18;
  ERR_TYPE_MISMATCH = -19;
  ERR_NOT_IMPLEMENTED = -20;
  ERR_ILLEGAL_OPERATION = -21;
  ERR_MISSING_OPERATION_TARGET = -22;
  ERR_CONFLICT_ON_WRITE_PRESERVE = -23;
  ERR_INACTIVE_TRANSACTION = -24;
  ERR_RESOURCE_LIMIT_REACHED = -25;
  ERR_DATA_CORRUPTION = -26;
  ERR_UNIQUE_CONSTRAINT_VIOLATION = -27;
}
